Advanced_Projects/PyFlowViz/parser/ast_parser.py

```
@@ -0,0 +1,319 @@
+                source_code = f.read()
+            
+            tree = ast.parse(source_code)
+            self._parse_ast(tree, file_path)
+            
+            return {
+                'nodes': list(self.nodes.values()),
+                'edges': self.edges,
+                'source_code': source_code
+            }
+        except Exception as e:
+            raise ValueError(f"Failed to parse file {file_path}: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)